src/DataGridExtensions/MultipleChoiceFilter.cs

```diff
@@ -229,7 +229,7 @@ private void ListBox_ItemsCollectionChanged(object? sender, NotifyCollectionChan
         {
             var filter = Filter;
 
-            if (filter?.Items == null)
+            if (filter?.Items == null && _listBox?.IsLoaded == true)

```

Comment:
Removed the double null check, and even though my testing didn't show a problem with the other SelectAll() calls, it might still be a good idea to check alright.